modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/logsdb/qa/StandardVersusLogsIndexModeChallengeRestIT.java

```diff
@@ -39,85 +44,71 @@
 import java.util.List;
 import java.util.Map;
 
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThan;
+
 public class StandardVersusLogsIndexModeChallengeRestIT extends AbstractChallengeRestTest {
+    private final DataGenerator dataGenerator;
 
     public StandardVersusLogsIndexModeChallengeRestIT() {
         super("standard-apache-baseline", "logs-apache-contender", "baseline-template", "contender-template", 101, 101);
+        this.dataGenerator = new DataGenerator(
+            DataGeneratorSpecification.builder()
+                // Nested fields don't work with subobjects: false.
+                .withNestedFieldsLimit(0)
+                // TODO increase depth of objects
+                // Currently matching fails because in synthetic source all fields are flat (given that we have subobjects: false)
+                // but stored source is identical to original document which has nested structure.
+                .withMaxObjectDepth(0)
+                .withArbitrary(new RandomBasedArbitrary() {
+                    // TODO enable null values
+                    // Matcher does not handle nulls currently
+                    @Override
+                    public boolean generateNullValue() {
+                        return false;
+                    }
+
+                    // TODO enable arrays
+                    // List matcher currently does not apply matching logic recursively
+                    // and equality check fails because arrays are sorted in synthetic source.
+                    @Override
+                    public boolean generateArrayOfValues() {
+                        return false;
+                    }
+                })
+                .withPredefinedFields(List.of(new PredefinedField("host.name", FieldType.KEYWORD)))
+                .build()
+        );
     }
 
     @Override
     public void baselineMappings(XContentBuilder builder) throws IOException {
         if (randomBoolean()) {
-            builder.startObject("properties")
-
-                .startObject("@timestamp")
-                .field("type", "date")
-                .endObject()
-
-                .startObject("host.name")
-                .field("type", "keyword")
-                .field("ignore_above", randomIntBetween(1000, 1200))
-                .endObject()
-
-                .startObject("message")
-                .field("type", "keyword")
-                .field("ignore_above", randomIntBetween(1000, 1200))
-                .endObject()
-
-                .startObject("method")
-                .field("type", "keyword")
-                .field("ignore_above", randomIntBetween(1000, 1200))
-                .endObject()
-
-                .startObject("memory_usage_bytes")
-                .field("type", "long")
-                .field("ignore_malformed", randomBoolean())
-                .endObject()
-
-                .endObject();
```

Comment:
I'd keep the hard-coded version as it's easier to debug, and add the randomized one as a separate test.